utils/secret_reader.py

```diff
@@ -0,0 +1,48 @@
+import utils.vault_client as vault_client
+
+
+def read(secret, settings=None):
+    """Returns a value of a key from Vault secret or configuration file.
+
+    The input secret is a dictionary which contains the following fields:
+    * path - path to the secret in Vault or config
+    * field - the key to read from the secret
+    * format (optional) - plain or base64 (defaults to plain)
+    * version (optional) - Vault secret version to read
+      * Note: if this is Vault secret and a v2 KV engine
+
+    The input settings is an optional app-interface-settings object
+    queried from app-interface.
+
+    Default vault setting is false, to allow using a config file
+    without creating app-interface-settings.
+    """
+
+    if settings and settings.get('vault'):
```

Comment:
Does this key need to be documented in the function description?  I don't see "vault" listed in the description of the fields in settings.

Reply:
The input settings is an optional app-interface-settings object queried from app-interface.

One of the keys is vault.

Reply:
added vault to the input settings description!